weasyl/journal.py

```diff
@@ -106,14 +106,53 @@ def select_view(userid, rating, journalid, ignore=True, anyway=None):
         "favorited": favorite.check(userid, journalid=journalid),
         "friends_only": "f" in journal.settings,
         "hidden_submission": "h" in journal.settings,
-        # todo
-        "fave_count": d.execute("SELECT COUNT(*) FROM favorite WHERE (targetid, type) = (%i, 'j')",
-                                [journalid], ["element"]),
+        "fave_count": favorite.count(journalid=journalid),
         "tags": searchtag.select(journalid=journalid),
         "comments": comment.select(userid, journalid=journalid),
     }
 
 
+def select_view_api(userid, journalid, anyway=False, increment_views=False):
+    rating = d.get_rating(userid)
+    db = d.connect()
+    journal = db.query(orm.Journal).get(journalid)
+    if journal is None or 'hidden' in journal.settings:
+        raise WeasylError('journalRecordMissing')
+    journal_rating = journal.rating.code
+    if 'friends-only' in journal.settings and not frienduser.check(userid, journal.userid):
+        raise WeasylError('journalRecordMissing')
+    elif journal_rating > rating and userid != journal.userid:
```

Comment:
Consistency nit: The character view api does a lot differently here as well. It checks for staff membership, SFW mode, etc. Is there a reason that they can't have the same behavior?